addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/ExtensionWebSocket.java

```diff
@@ -535,6 +551,58 @@ public String getDescription() {
         return Constant.messages.getString("websocket.desc");
     }
 
+    private void registerUserScripts(ScriptType scriptType) {
+        String folder = SCRIPTS_FOLDER + scriptType.getName() + '/';
+        Path scriptFolderPath = Paths.get(folder);
+        try (Stream<Path> scriptFilePaths = Files.list(scriptFolderPath)) {
+            scriptFilePaths
+                    .map(path -> path.toFile())
+                    .filter(file -> file.isFile()) // Keep only those which are files
+                    .map(
+                            file -> {
+                                return new ScriptWrapper(
+                                        file.getName(),
+                                        "",
+                                        extensionScript.getEngineNameForExtension(
+                                                file.getName()
+                                                        .substring(
+                                                                file.getName().lastIndexOf(".")
+                                                                        + 1)),
+                                        scriptType,
+                                        true,
+                                        file);
+                            })
+
+                    // Load the scripts from ExtensionScript(if not an IOException)
+                    .map(scriptWrapper -> loadScript(scriptWrapper))
+                    // Check if they are null
+                    .filter(maybeScriptWrapper -> maybeScriptWrapper.isPresent())
+                    // Get the ScriptWrapper
+                    .map(maybeScriptWrapper -> maybeScriptWrapper.get())
+                    .filter(
+                            scriptWrapper ->
+                                    this.extensionScript.getScript(scriptWrapper.getName()) == null)
+                    .forEach(scriptWrapper -> this.extensionScript.addScript(scriptWrapper, false));
+
+        } catch (NoSuchFileException e) {
+            return;
+        } catch (IOException e) {
+            logger.warn(
+                    "A problem occurred while trying to import scripts for " + scriptType.getName(),
+                    e);
+            e.printStackTrace();
```

Comment:
Should use `logger` or just be removed (seems handled on previous line).

Reply:
Fixed